streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -17,117 +17,231 @@
 package org.apache.kafka.streams.state.internals.metrics;
 
 import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Utils;
 import org.apache.kafka.streams.processor.TaskId;
 import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
 import org.apache.kafka.streams.state.internals.metrics.RocksDBMetrics.RocksDBMetricContext;
 import org.rocksdb.Statistics;
 import org.rocksdb.StatsLevel;
+import org.rocksdb.TickerType;
+import org.slf4j.Logger;
 
-import java.util.HashMap;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class RocksDBMetricsRecorder {
+    private Logger log;
 
     private Sensor bytesWrittenToDatabaseSensor;
-    private Sensor bytesReadToDatabaseSensor;
+    private Sensor bytesReadFromDatabaseSensor;
     private Sensor memtableBytesFlushedSensor;
     private Sensor memtableHitRatioSensor;
-    private Sensor memtableAvgFlushTimeSensor;
-    private Sensor memtableMinFlushTimeSensor;
-    private Sensor memtableMaxFlushTimeSensor;
     private Sensor writeStallDurationSensor;
     private Sensor blockCacheDataHitRatioSensor;
     private Sensor blockCacheIndexHitRatioSensor;
     private Sensor blockCacheFilterHitRatioSensor;
     private Sensor bytesReadDuringCompactionSensor;
     private Sensor bytesWrittenDuringCompactionSensor;
-    private Sensor compactionTimeAvgSensor;
-    private Sensor compactionTimeMinSensor;
-    private Sensor compactionTimeMaxSensor;
     private Sensor numberOfOpenFilesSensor;
     private Sensor numberOfFileErrorsSensor;
 
-    final private Map<String, Statistics> statisticsToRecord = new HashMap<>();
-    final private String metricsScope;
-    final private String storeName;
+    private final Map<String, Statistics> statisticsToRecord = new ConcurrentHashMap<>();
+    private final String metricsScope;
+    private final String storeName;
+
+    private enum State { NEW, RUNNING, NOT_RUNNING, ERROR, MANUAL }
+    private volatile State state = State.NEW;
+    private final Duration recordingInterval = Duration.ofMinutes(10);
+    private final boolean startRecordingThread;
+    private Thread thread;
 
     public RocksDBMetricsRecorder(final String metricsScope, final String storeName) {
+        this(metricsScope, storeName, true);
+    }
+
+    // visible for testing
+    RocksDBMetricsRecorder(final String metricsScope, final String storeName, final boolean startRecordingThread) {
```

Comment:
nit: the last parameter seems always true in non-testing code. I'm wondering if we can add a couple `test-only` functions (e.g. the following `error()` seems test only also), like a `setManual` which would then force the state to be MANUAL instead of setting it at the constructor.

Then we can remove the `startRecordingThread` flag, and in tests after we've created the recorder object call `setManual` so that later `add/removeStatistics` would not start or stop recording.